### ISSUE TITLE
remove optimism feature from chainspec crate

### ISSUE THREAD

**Issue by mattsse** [paradigmxyz/reth#9450]
### Describe the feature

blocked by #9439

once we have separate binaries we can now also get rid of the optimism features we have in chainspec crate.

the duplicated is_optimism function can be removed because not called by eth stuff anymore.
and DEV spec can be duplicated to OP_DEV and regular `DEV`

**Comment by loocapro** [comment]
May I?

**Comment by emhane** [comment]
duplicate of https://github.com/paradigmxyz/reth/issues/8904
